Cargo.toml

```diff
@@ -25,7 +25,9 @@ at-commands = "0.5.2"
 no-std-net = "0.6.0"
 critical-section = "1.1"
 embedded-io = { version = "0.4.0", features = ["async"] }
+heapless = { version = "0.7.16", optional = true }
 
 [features]
 default = []
 defmt = ["dep:defmt", "at-commands/defmt"]
+sms = ["heapless"]
```

Comment:
If you're able to replace the formatting code (which I put a comment on with a suggestion), then I don't think we need to feature gate this

Reply:
I removed the feature gate for now. But the write! As said above is still present